### DISCUSSION TITLE
sosa:usedProcedure

### DISCUSSION THREAD

**Discussion by GrantATgaia** [ternaustralia/ontology_tern#75] [Q&A]
Is sosa:usedProcedure mandatory?

We may not have this declared every WA BIO dataset. 

Can we relax that as a mandatory requirement?

**Comment by smguru** [comment]
usedProcedure provides a procedure used to make an observation. In O&M the core properties are observable property, procedure, phenomenon time and result. probably it will be good to keep the procedure to comply with the standard but we can relax the constraint for usedProcedure to accept text as well so there is no need to create a vocabulary for the procedure used in any observation.
Probably @dr-shorthair and @edmondchuc can comment as well

**Comment by dr-shorthair** [comment]
`sosa:usedProcedure` is NOT mandatory - see https://www.w3.org/TR/vocab-ssn/#SOSAObservation 

In fact nothing is 'mandatory' in the SSN/SOSA ontology. Cardinality constraints just say how many are _expected_. OWL/RDFS does not address data 'validity', only entailments - i.e. what can be _inferred_ from the information provided. So if cardinality=1 and there are two values provided, then it will be inferred that these are the same thing (which may be 'inconsistent' - which is not the same as 'invalid'). 

OTOH - SHACL does deal with 'validity'. SSN/SOSA does not have any SHACL. TERN Ontology does, which in some cases adds some constraints to the semantics originally provided by the SSN/SOSA Ontology.

**Comment by miekeGR** [comment]
Hi @edmondchuc I've been looking at the worked example https://github.com/ternaustralia/bdr-faealla-worked-example/commit/ffcb897a8374c0eef7f79641706e990a5c0e74d5 and see that for the sosa:usedProcedure lines have links to a tbc controlled vocabulary, but I think we need further discussion on how we implement this for the BDR. @kheerand can we add this to the agenda for our next meeting?

**Reply by edmondchuc** [reply]
@miekeGR one of our strategies with providing procedures is to look at the original PDF document and take out the relevant steps and republish it as its own entity online. Example, this [genetic isotope and sample vouchering method](http://linked.data.gov.au/def/ausplots-cv/a526b5c5-5931-4460-ba17-68c0fcc3e710) from the AusPlots Rangelands protocol.

I am not sure if this is the kind of detail what Gaia Resources or the BDR are looking to do. Perhaps simply linking to an online version of the entire survey protocol is enough? I'm not sure. 

Another option would be to have a `Procedure` class where it has a link to the entire survey protocol while allowing for comments such as, "see page 31".

**Reply by miekeGR** [reply]
Thanks for the suggestion @edmondchuc . I think we are tossing up ideas around using a controlled vocabulary of fairly generic terms here for the BDR. Would that be acceptable?

**Reply by edmondchuc** [reply]
Hmm, I don't think describing procedures in a generic way is suitable. 

For example, in ecology, most datasets collect the diameter of a tree at breast height (DBH). Depending on the protocol, the diameter may always be measured at 1.5 m, but some protocols may specify the breast height as a different value. There are usually other instructions as well which affect when and how they measure a tree.

Below is a brief snippet on the procedure of how DBH is measured from the AusPlots Forests survey protocol [1].

> Record the diameter of each tree ≥10cm at 1.3m, unless they are a ‘problem tree’ (i.e. leaning, on a slope, 
buttressed etc. see Solutions to Problem Trees Box). Use a measuring pole pushed firmly into the leaf litter to 
define 1.3m. Pull the diameter tape around the tree trunk such that it is perpendicular to the main axis of the 
tree trunk (i.e. not parallel to the ground). Clear any moss, loose bark or anything else that might distort the 
diameter tape. 

They go into more detail on how DBH is measured if a tree is determined to be a "problem tree".

---

[1] - page 31 of [AusPlots Forests survey protocols manual](https://www.tern.org.au/wp-content/uploads/TERN-Forests-Field-Manual_web.pdf)
